pkg/kube/activity.go

```diff
@@ -51,6 +52,37 @@ type PipelineDetails struct {
 	Build         string
 }
 
+var disallowedNameChars = regexp.MustCompile("[^a-z0-9-]")
```

Comment:
it is not uncommon for branches (or owners or repsoitories) to contain upper case (or in the case of branches be only upper case?  do we not allow these for some reason (if so could you say what it is?  (and if so possibly replacing all the `UPPER` case with `-lower` may be better - example Org/MySuperProject/JIRA-1234 -> `-org--my-super-project--j-i-r-a-1234` rather than `-rg--y-uper-roject/----1234`   (esp if you have multiple JIRA projects for a shared repo that have the same lengh names)

Reply:
Agreed - no reason not to allow upper case here - I'll fix it so we keep them, it wasn't intended to turn them all into '-'.